### PR TITLE
feat: t6 - get only dirs of the files

### PR DESCRIPTION
**Pull request by KrawczowaKris**
# Summary
## Test Task
**T6 - Get only dirs of the files**
Script should get a folder as an argument and return a list of unique directories of all files (including in the nested directories). only -f option is allowed for the use of find command, i.e.: find /tmp -type f. Which should find all files in all nested dirs and then process the output by using expansion.

## my results
`./chapter2-l2-t6-get-file-unique-dirs.sh ~/fig`

**output:**
```
/home/nadya/fig
/home/nadya/fig/qw
/home/nadya/fig/fds
/home/nadya/fig/fds/rert
/home/nadya/fig/sss
```

### CODE REVIEW

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (kseniyashaydurova) [inline chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:1]: Script works incorrectly
image

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+files=$(find $folder -type f)
+dirs=( )
+
+for file in $files; do
+  dir="${file%/*}"
+  if [[ ! "${dirs[0]}" =~ $dir ]]; then
+    dirs+=$dir
+    echo $dir
+  fi
+done
```
Comment (kseniyashaydurova) [inline chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:16]: Please always add 1 empty line at the end of each script

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Script get a folder as an argument and return a list of unique directories
+# of all files
+
+folder=$1
```
Comment (kseniyashaydurova) [inline chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:6]: Please also force users to pass only 1 argument

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh (hunk 2 above)
Comment (KrawczowaKris) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:16]: done

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh (hunk 3 above)
Comment (KrawczowaKris) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:6]: done

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+files=$(find $folder -type f)
+dirs=( )
```
[the remark this answers is not in the document]
Comment (KrawczowaKris) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:8]: done

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh (hunk 1 above)
Comment (KrawczowaKris) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:1]: done

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**